### PR TITLE
[SHELL32] Improve StartButton context menu

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

JIRA issue: [CORE-16544](https://jira.reactos.org/browse/CORE-16544)

## Proposed changes

- Add `uFlags` argument to `CDefaultContextMenu::AddShellExtensionsToMenu`.
- `CCopyToMenu`, `CMoveToMenu`, and `CSendToMenu` check the `uFlags` against `(CMF_NOVERBS | CMF_VERBSONLY)`.

## TODO

- [x] Do tests.

## Screenshots

BEFORE:
before-1

AFTER:
after-1

### CODE REVIEW

File: dll/win32/shell32/CSendToMenu.cpp
Comment (JoachimHenze) [inline dll/win32/shell32/CSendToMenu.cpp:298]: If you believe your PR to be a proper solution I would recommend to strip the mentioning of CORE-16544 here and at the other places where you added that.
We do use such comments in the code only for temporary hacks and workarounds. (If you believe your PR would fall into that category, then it would be okay to keep them, but then the PRs title/description should be changed to reflect it being a hack.)
For now I do assume your PR is a proper fix. (?!?)

File: dll/win32/shell32/CCopyToMenu.cpp
Comment (JoachimHenze) [inline dll/win32/shell32/CCopyToMenu.cpp:277]: I would omit the opening and closing curly braces for conditions where the commands do need only a single line.
Here and also at the other 2 places where such braces were added. That style is a giant waste of vertical space in a source file. Those can be added whenever a 2nd command would get added. And no: Developers won't forget that: There is static code analysis, we are not in the 80ies anymore. ;-)

File: dll/win32/shell32/CCopyToMenu.cpp
Comment (JoachimHenze) [inline dll/win32/shell32/CCopyToMenu.cpp:274]: != 0
 Can be stripped here and at the other 2 locations.

**Comment by JoachimHenze** [comment]
I just tested: Our unpatched explorer does not show the erroneous SendTo-menu when being executed on XPSP3. I cannot say whether that renders this PR a hack or not, because it does changes in both: explorer.exe and shell32.dll.
And I used only explorer.exe for the cross-test, but did not replace shell32.dll as well, so I was testing with XPSP3 shell32.dll.
Maybe others can deduce better than myself whether that outcome does strengthen or weaken the trust in this PRs correctness.

**Comment by katahiromz** [comment]
I have reverted `explorer`.

**Review by JoachimHenze — APPROVED** [review]
LGTM.
Wait until 3 approvals (without rejection) for immediate merge.
Merge in one week otherwise.

**Review by HBelusca — APPROVED** [review]

**Pull request merged**
